integration/shareddata/shareddata.go

```diff
@@ -242,8 +242,85 @@ func (b *BenchmarkValues) Docs() iterator.Interface[struct{}, bson.D] {
 	return b.iter
 }
 
+// field represents a field in a document.
+type field struct {
+	Value any
+	Key   string
+}
+
+// Fields is a map of field name -> value.
```

Comment:
```suggestion
// Fields is a slice of ordered field name value pair. To avoid fields being inserted in different order between compat and target, use a slice instead of a map.
```